docs/developer-docs/production/best-practices.md

```diff
@@ -0,0 +1,60 @@
+# Development Best Practices 
+
+## Overview
+
+This guide outlines essential development best practices for building on the Internet Computer Protocol (ICP).
+
+## Best Practices
+
+### 1. Generous Canister Top-Up
+
+Ensure all canisters have sufficient cycles to sustain operations for several years initially. Given the cost-effectiveness of storage and computation on ICP compared to other platforms, this upfront investment is usually modest.
+
+Refer to [topping up cycles page](../production/topping-up-canister.md) to learn how to top up cycles effectively.
+
+Refer to the [cycles estimate](../gas-cost.md) page to estimate cycle usage. 
+
+### 2. Managing Cycle Depletion
+
+ICP features a mechanism to prevent canisters from running out of cycles. Canisters have a configurable `freezing_threshold`, dynamically evaluated in cycles. Set `freezing_threshold` conservatively, especially for NFTs, ensuring at least 90 to 180 days' worth of cycles for proactive management.
+
+Refer to [this resource](../../tutorials/developer-journey/level-1/1.6-managing-canisters.md/#setting-freezing-threshold) to learn how to configure the `freezing_threshold` of a canister. 
```

Comment:
This link into the middle of a developer journey might be confusing. Maybe we could copy that explanation into topping-up-canister.md ?

Reply:
Thank you for the feedback! Done here!